test/sharness/t0252-files-gc.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+#
+# Copyright (c) 2016 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="test how the unix files api interacts with the gc"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+
+test_expect_success "object not removed after gc" '
+  echo "hello world"  | ipfs files write --create /hello.txt &&
+  ipfs repo gc &&
+  ipfs cat QmVib14uvPnCP73XaCDpwugRuwfTsVbGyWbatHAmLSdZUS
```

Comment:
So... if this is "best effort", under what conditions should gc remove these, and can we make sure to test that here?